src/js/jsx/search/SearchBar.jsx

```diff
@@ -250,41 +303,232 @@ define(function (require, exports, module) {
 
             var recentDocLabel = {
                 id: "recent-doc_header",
-                title: "Recent Documents",
+                title: HEADERS.RECENT_DOCS,
                 type: "header"
             };
 
-            recentDocMap.unshift(recentDocLabel);
+            return recentDocMap.unshift(recentDocLabel);
+        },
+
+        /**
+         * Make list of search category dropdown options based on header
+         * 
+         * @param {string} header, either "layer" or "document"
+         * @return {Array.<object>}
+         */
+        _getFilterOptions: function (header) {
+            if (header !== "layer" && header !== "document") {
+                return;
```

Comment:
This should either throw an exception or just be removed. Returning here will just cause a failure elsewhere. An explicit exception is fine, but also for private functions we generally don't feel like it's necessary to do error checking like this up front unless the correctness conditions of the parameters are obscure.